src/core/lib/iomgr/executor.cc

```diff
@@ -264,6 +264,9 @@ void Executor::ThreadMain(void* arg) {
     grpc_core::ExecCtx::Get()->InvalidateNow();
     subtract_depth = RunClosures(ts->name, closures);
   }
+#if !defined(__APPLE__)  // because apple OSes do not shutdown executor
```

Comment:
This isn't true. Only iOS doesn't shut down the executor, and MacOS seems to work fine with the TLS change. Can you just limit this to iOS?

Reply:
This does happen in MacOS as well. Please see https://github.com/grpc/grpc/pull/19978